src/openforms/js/components/formio_builder/WebformBuilder.js

```diff
@@ -33,6 +34,80 @@ class WebformBuilder extends WebformBuilderFormio {
+            // 'label': the label of the component and the labels of the options. From within the
+            // this.editForm.everyComponent function it didn't seem possible to differentiate which component was being
+            // edited. So, to prevent changes to the options labels to reset the key, a custom option "isOptionLabel"
+            // was added to them in src/openforms/js/components/form/edit/tabs.js.
+            if (!event.changed) return;
+
+            if (event.changed.component && event.changed.component.key === 'showFullSchema') {
+                const { value } = event.changed;
+                this.editForm.submission = {
+                    data: {
+                        componentJson: component,
+                        showFullSchema: value
+                    },
+                };
+                return;
+            }
+
+            // See if this is a manually modified key. Treat custom component keys as manually modified
+            if ((event.changed.component && (event.changed.component.key === 'key')) || isJsonEdit) {
+                componentCopy.keyModified = true;
+            }
+
+            if (event.changed.component && (['label', 'title'].includes(event.changed.component.key))) {
+                // Ensure this component has a key.
+                if (isNew) {
+                    if (!event.data.keyModified && !event.changed.component.isOptionLabel) {
+                        this.editForm.everyComponent(component => {
+                            if (component.key === 'key' && component.parent.component.key === 'tabs') {
+                                component.setValue(_.camelCase(
+                                    event.data.title ||
+                                    event.data.label ||
+                                    event.data.placeholder ||
+                                    event.data.type
+                                ).replace(/^[0-9]*/, '') );
+
+                                return false;
+                            }
+                        });
+                    }
+
+                    if (this.form) {
+                        let formComponents = this.findNamespaceRoot(parent.formioComponent);
+                        // excluding component which key uniqueness is to be checked to prevent the comparing of the same keys
+                        formComponents = formComponents.filter(
+                            comp => this.editForm.options.editComponent.id !== comp.id
+                        );
+
+                        // Set a unique key for this component.
+                        BuilderUtils.uniquify(formComponents, event.data);
+                    }
+                }
+            }
+
+            // Update the component.
+            this.updateComponent(event.data.componentJson || event.data, event.changed);
+        };
+
+        this.editForm.events._events['formio.change'][existingOnChangeHandlers.length-1].fn = modifiedOnChangeCallback;
```

Comment:
The only thing I'm a bit worried about is that this is index-based to patch the library handler, but I don't think we have any better alternatives. It's something to keep in mind if/when we upgrade our Formio version.